### PR TITLE
assorted fixes relating to #130

### PR DESCRIPTION
**Pull request by ossilator**
so of course i found more stuff once is started looking at it ...

this time i actually tested the error paths as well, by provoking xruns with strategically placed delays.

i think the docu updates are adequate. i believe that there are no other relevant sections that need adjustment.

i did not bother adjusting the examples, as they are generally of questionable quality, and it sort of felt like securing the door of a blown up house. i think i'll make a separate PR with related improvements.

please review and approve this PR, but do **NOT** merge it - i want to do that myself, to see if everything is set up to my liking. EDIT: i think it looks good - the mode i want applied is "rebase and merge", to preserve history, cleanly.

### CODE REVIEW

File: alsaaudio.c [hunk 1]
```diff
@@ -1463,45 +1473,48 @@ static PyObject *alsapcm_write(alsapcm_t *self, PyObject *args)
...
+		}
 	}
 
 #if PY_MAJOR_VERSION >= 3
 	PyBuffer_Release(&buf);
 #endif
 
-	return rc;
+	return PyLong_FromLong(res);
 }
```
Comment (larsimmisch) [inline alsaaudio.c:1518]: Ouch.

**Review by larsimmisch — APPROVED** [review]
This looks very good to me, thank you. 

I'll try to find time to test this in the next couple of days on my usecase (and re-read the documentation). 

Feel free to merge yourself (my default github setup is on _squash and merge_ for unrelated reasons).

**Comment by ossilator** [comment]
with the history now being legible, i was actually able to figure out what exactly went wrong and when. as a consequence, there are more commits now.
also, rebased on rewritten main, obviously.
the diff to the previous state is empty, sans one excessive changelog entry (the bug was introduced post-release).

**Comment by ossilator** [comment]
> my default github setup is on squash and merge for unrelated reasons

do these reasons still apply, now that we kind of have a proper commit and review policy?
i'd change it, to avoid this mistake in the future.

**Comment by larsimmisch** [comment]
Thanks!

**Pull request merged**